File: matiasmasso/Mat.NET/DT2/DTOAreaRegio.vb
Public Class DTOAreaRegio
    Inherits DTOArea
    Shadows Property Country As DTOCountry
    Property Provincias As List(Of DTOAreaProvincia)

    Public Sub New()
        MyBase.New()
        _Provincias = New List(Of DTOAreaProvincia)
    End Sub

    Public Sub New(oGuid As Guid, Optional sNom As String = "")
        MyBase.New(oGuid, sNom)
        _Provincias = New List(Of DTOAreaProvincia)
    End Sub

    Shared Shadows Function Factory(oCountry As DTOCountry) As DTOAreaRegio
        Dim retval As New DTOAreaRegio
        With retval
            .Country = oCountry
        End With
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOAreaRegio.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOAreaRegio : DTOArea
    {
        public new DTOCountry Country { get; set; }
        public List<DTOAreaProvincia> Provincias { get; set; }

        public DTOAreaRegio() : base()
        {
            Provincias = new List<DTOAreaProvincia>();
        }

        public DTOAreaRegio(Guid oGuid, string sNom = "") : base(oGuid, sNom)
        {
            Provincias = new List<DTOAreaProvincia>();
        }

        public static DTOAreaRegio Factory(DTOCountry oCountry)
        {
            DTOAreaRegio retval = new DTOAreaRegio();
            {
                var withBlock = retval;
                withBlock.Country = oCountry;
            }
            return retval;
        }
    }
}
